src/Quiz.js

```diff
@@ -1,17 +1,110 @@
+import { Button, Card, Radio } from 'antd';
+import Score from './Score.js';
+import shuffle from 'shuffle-array';
+import InnerHTML from 'dangerously-set-inner-html';
+
+const RadioButton = Radio.Button;
+const RadioGroup = Radio.Group;
+
+class Quiz extends Component {
+  constructor(props){
+    const { quizId } = props.match.params;
+    super()
+    this.state={
+      questionsData: JSON.parse(window.localStorage.getItem(quizId)),
+      disabledQuestion:false,
+      streakBar: 0,
+      progressBar: [],
+    }
+  }
+
+  check = (choice, answer, quizId, questionId, lastQuestion) => {
+    this.checkScoreAnswer(choice, answer);
+    alert(`you picked ${choice}, the correct answer is${answer}`)
+    this.nextQuestion(quizId, questionId, lastQuestion);
+  }
+
+  nextQuestion = (quizId, questionId, lastQuestion) => {
+    if (questionId < lastQuestion){
+      this.props.history.push(`/quizzes/${quizId}/questions/${questionId + 1}`)
+    }
+  }
+
+  onFetchQuestions = (data) => {
+    this.setState({questionsData: data.results});
+  }
+
+  checkScoreAnswer = (choice, answer) => {
+    choice===answer? this.updateProgressBar():this.resetStreakBar();
+  }
+
+  resetStreakBar = ()=>{
+    const currentProgressBar = this.state.progressBar;
+    this.setState(({streakBar})=>({streakBar: 0}));
+    currentProgressBar.push(false)
+    this.setState(({progressBar})=>({progressBar: currentProgressBar}));
+  }
+  updateStreakBar = ()=>{
+    this.setState(({streakBar})=>({streakBar: this.state.streakBar +1 }));
+  }
+
+  updateProgressBar = ()=>{
+    const currentProgressBar = this.state.progressBar;
+    currentProgressBar.push(true)
+    this.setState(({progressBar})=>({progressBar: currentProgressBar}));
+    this.updateStreakBar();
+  }
+
+  render(){
+    const { match } = this.props;
+    const quizId = match.params.quizId;
+    const idquest = Number(match.params.id);
+    const prevQuestion = idquest - 1;
+    const nextQuestion = idquest + 1;
+    const lastQuestion = this.state.questionsData.length - 1;
+    const quiz = this.state.questionsData[idquest]
+    const title = quiz.question;
+    const choices = shuffle(quiz.incorrect_answers.concat(quiz.correct_answer));
+    const answer = quiz.correct_answer;
+    return (
+      <div className='quiz-content'>
+        <Score streakBar={this.state.streakBar} results={this.state.progressBar}/>
+        <Card
+            className='quiz-card'
+            match={match}
+            title={<InnerHTML html={title}/>}
+          >
+            <RadioGroup defaultValue="a">
+              {choices.map((choice, i) => {
+                return <RadioButton
+                  key={i}
+                  disabled={this.state.disabledQuestion}
+                  onClick={()=>this.check(choice, answer, quizId, idquest, lastQuestion)}
+                  style={{display:'block'}}
+                  value={choice}
+                >
+                  {<InnerHTML html={choice}/>}
```

Comment:
You don't need interpolators here `{}` and dangerouslySetInnerHtml is a property on all native element types. No need for the package.

https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml